bundles/specmate-auth/src/com/specmate/auth/internal/PersistentSessionService.java

```diff
@@ -111,10 +123,15 @@ public AccessRights getTargetAccessRights(String token) throws SpecmateException
 	}
 
 	@Override
-	public void delete(String token) throws SpecmateException {
-		UserSession session = (UserSession) sessionTransaction.getObjectById(getSessionID(token));
-		SpecmateEcoreUtil.detach(session);
-		sessionTransaction.commit();
+	public void delete(String token) throws SpecmateException, SpecmateValidationException {
+		sessionTransaction.doAndCommit(new IChange<Object>() {
+			@Override
+			public Object doChange() throws SpecmateException, SpecmateValidationException {
+				UserSession session = (UserSession) sessionTransaction.getObjectById(getSessionID(token));
```

Comment:
you could pull this line out, also
BTW: What would happen if the the object with the id is not found?